docs/workers.md

```diff
@@ -1,18 +1,54 @@
 Marathon Workers
 ================
+## Overview
 
-## Create CSV From Filters Worker
+Depending on the job type, different workers can be called. Se the image below:
 
-This worker queries the PUSH_DB using the job filters and builds a CSV file containing user ids that will receive this push notification. Finally, it uploads this CSV file to AWS S3 and calls the next worker (create batches from csv worker).
+![example](img/Workers.png)
 
-## Create Batches From CSV Worker
+This image shows the order workers execution order. Also, you can see each data that go between the workers.
 
-This worker downloads a CSV file from AWS S3, reads it and creates batches of user information (locale, token, tz) grouped by timezone. If a job is scheduled and not localized, it schedule all batches in the next worker (process batch worker) for the same timestamp. If a job is scheduled and localized it schedules each batch according to the corresponding timestamp for each timezone. If a job is not schedule it calls the next worker directly for each batch.
+When we have a scheduled job with multiple timezones, multiple jobs will be created and scheduled. Each job will be processed in 1 hour interval, tats mean, if a user is at the +1215 timezone he will receive a message when the +1200 is processed.
 
-## Process Batch Worker
+## Workers 
 
-This worker receives a batch of user information (locale and token), builds the template for each user using the locale information and the job template name and send to the kafka topic corresponding to the job app and service. If the error rate is more than a threshold this job enters circuit break state. When the job is paused or in circuit break the batches are stored in a paused job list in Redis with an expiration of one week.
+### Direct Worker
 
-## Resume Job Worker
+This worker receives from the API batches database intervals to process. It will query the PUSH_DB using the job filters, creates the messages and send to Kafka. This worker is really fast and can handle big amount of tokens (tested with 1.5x10^8 tokens).
+
+If the control group is set, it will save it on Redis. The completed job worker will pull this data and create a CSV for it.
+
+It will not generate a CSV of the sent messages.
+
+This worker will produce two metrics: `starting_direct_part` that represent when it receives some data. After getting some data from the DB, it will send the timing metric `get_from_pg` representing the time spend to retrieve the data.
```

Comment:
This is quite confusing. Maybe split in topics:

This worker will produce two metrics:
- `starting_direct_part`: ...
- `get_from_pg`: ...